colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -117,29 +121,72 @@ class Collection(val config: CollectorConfig) {
         )
       }
     }
-    if (collectors.containsKey(address)) {
-      cast(collectors.get(address))
-    } else {
+    addContext(ns)
```

Comment:
If a metric already exists at the address, does this end up overwriting whatever tags are set without changing the collector?

It might make more sense to have a single var (which will be an atomic reference) that stores a tuple of a collector and a tagmap, and both are set at the same time.